nixos/lib/test-driver/test_driver/tpm.py

```diff
@@ -0,0 +1,54 @@
+import subprocess
+from tempfile import TemporaryDirectory
+
+
+class Tpm:
+    """
+    This is a TPM driver for QEMU tests.
+    It gives you access to a TPM socket path on the host
+    you can access at `tpm_socket_path`.
+    """
+
+    state_dir: TemporaryDirectory
+    swtpm_binary_path: str
+    tpm_socket_path: str
+
+    def __init__(self, swtpm_binary_path: str, tpm_socket_path: str):
+        self.state_dir = TemporaryDirectory()
+        self.swtpm_binary_path = swtpm_binary_path
+        self.tpm_socket_path = tpm_socket_path
+        self.start()
+
+    def start(self) -> None:
+        """
+        Start swtpm binary and wait for its proper startup.
+        In case of failure, this will raise a runtime error.
+        """
+        self.proc = subprocess.Popen(
+            [
+                self.swtpm_binary_path,
+                "socket",
+                "--tpmstate",
+                f"dir={self.state_dir.name}",
+                "--ctrl",
+                f"type=unixio,path={self.tpm_socket_path}",
+                "--tpm2",
+            ]
+        )
+
```

Comment:
I'd poll for `self.tpm_socket_path` to appear for one or two seconds. Just to make sure we don't hand that over to qemu before it's ready.

Reply:
`swtpm` does have a `--daemon` flag. I wonder if it does that correctly, where the parent process only exits successfully once it knows the child is running and ready for use. If so, rather than polling, and rather than the little `0.2` second timeout below, we could just use `--daemon` and wait for a successful exit code from the parent.